scripts/derive_galileo_gas_parameter/show_batches.py

```
@@ -0,0 +1,81 @@
+from pathlib import Path
+
+def show_batches(filename):
+    """Show batch IDs in the data file"""
+
+    if not Path(filename).exists():
+        print(f"Error: File not found: {filename}")
+        return
+
+    try:
+        with open(filename, 'rb') as f:
+            data = pickle.load(f)
```

Ruff (0.15.2)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)